src/components/Reports/Reports.js

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import ReportsFilters from './ReportsFilters/ReportsFilters';
+import { getDomains, getPagesByDomainId } from './ReportsService';
+
+class Reports extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      domains: [],
+      domainSelected: null,
+      pages: [],
+      pageSelected: null,
+    };
+
+    this.changeDomain = this.changeDomain.bind(this);
+  }
+
+  async componentDidMount() {
+    const domains = await getDomains();
+    if (domains.length) {
+      const domainSelected = domains[0];
+      const pages = await getPagesByDomainId(domainSelected.id);
+      const pageSelected = pages.length ? pages[0] : null;
+      this.setState({
+        domains: domains,
+        domainSelected: domainSelected,
+        pages: pages,
+        pageSelected: pageSelected,
+      });
+    }
+  }
+
+  changeDomain = async (event) => {
+    const id = parseInt(event.target.value);
+    const domainFound = this.state.domains.find(function(item) {
+      return item.id === id;
+    });
+    const pages = await getPagesByDomainId(id);
+    const pageSelected = pages.length ? pages[0] : null;
+    this.setState({ domainSelected: domainFound, pages: pages, pageSelected: pageSelected });
+  };
+
+  render() {
+    return (
+      <>
+        <ReportsFilters
+          changeDomain={this.changeDomain}
+          domains={this.state.domains}
+          domainSelected={this.state.domainSelected}
+          pages={this.state.pages}
+        />
```

Comment:
Since you are updating the state with _pageSelected_, I think that you should pass it here and make honor to it in ReportsFilter.

Reply:
fixed